src/gwt/src/org/rstudio/studio/client/common/mathjax/MathJax.java

```diff
@@ -300,12 +317,14 @@ private void renderLatexLineWidget(final Range range,
             @Override
             public void execute(LineWidget w)
             {
+               pendingLineWidgets_.remove(row);
```

Comment:
Ah, I wasn't aware that these `PinnedLineWidget` objects would exist + be bound to a row before the `getLineWidgetAtRow()` API would be able to provide widgets for that row. That will definitely be cleaner; thanks!